### PR TITLE
br: support access by azure sas token

### PR DESCRIPTION
**Pull request by Leavrth**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

support access blob by [SAS (shared access signatures)](https://learn.microsoft.com/en-us/rest/api/storageservices/delegate-access-with-shared-access-signature).
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
Could you please invite a tech reviewer? Thanks!

**Review by YuJuncen — APPROVED** [review]

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -185,11 +186,15 @@ BACKUP DATABASE * TO 's3://bucket-name/prefix' SEND_CREDENTIALS_TO_TIKV = FALSE;
 </div>
 <div label="Azure Blob Storage" value="azure">
 
-- 方式一：指定访问密钥
+- 方式一：指定共享访问签名
+
+    在 URI 配置 `account-name` 和 `sas-token`，则使用该参数指定的共享访问签名。由于共享访问签名中带有 `&` 的字符，需要在编码为 `%26` 后，再添加到 URI 中。你也可以直接对整个 sas-token 进行一次编码。
```
Comment (3pointer) [inline br/backup-and-restore-storages.md:191]: 可以给个样例

Comment (Leavrth) [reply br/backup-and-restore-storages.md]: add todo

File: br/backup-and-restore-storages.md [hunk 2]
```diff
@@ -185,11 +186,15 @@ BACKUP DATABASE * TO 's3://bucket-name/prefix' SEND_CREDENTIALS_TO_TIKV = FALSE;
...
 <div label="Azure Blob Storage" value="azure">
 
-- 方式一：指定访问密钥
+- 方式一：指定共享访问签名
+
+    在 URI 配置 `account-name` 和 `sas-token`，则使用该参数指定的共享访问签名。由于共享访问签名中带有 `&` 的字符，需要在编码为 `%26` 后，再添加到 URI 中。你也可以直接对整个 sas-token 进行一次编码。<!-- TODO: add an example -->
+
+- 方式二：指定访问密钥
 
     在 URI 配置 `account-name` 和 `account-key`，则使用该参数指定的密钥。除了在 URI 中指定密钥文件外，还支持 br 命令行工具读取 `$AZURE_STORAGE_KEY` 的方式。
```
Comment (Oreoxmt) [inline br/backup-and-restore-storages.md:195]:
```suggestion
    在 URI 配置 `account-name` 和 `account-key`，则使用该参数指定的存储账户名和密钥。除了在 URI 中指定密钥文件外，还支持 br 命令行工具读取 `$AZURE_STORAGE_KEY` 的方式。
```

File: br/backup-and-restore-storages.md [hunk 3]
```diff
@@ -76,7 +76,8 @@ BACKUP DATABASE * TO 's3://bucket-name/prefix' SEND_CREDENTIALS_TO_TIKV = FALSE;
 
     - `account-name`：存储账户名
     - `account-key`：访问密钥
-    - `access-tier`：上传对象的存储类别，例如 `Hot`、`Cool`、`Archive`，默认为 `Hot`
+    - `sas-token`：共享访问签名
```
Comment (Oreoxmt) [inline br/backup-and-restore-storages.md:79]:
```suggestion
    - `sas-token`：共享访问签名令牌
```

**Review by qiancai — APPROVED** [review]
/lgtm

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**